internal/services/mssql/mssql_server_resource.go

```diff
@@ -396,8 +396,8 @@ func resourceMsSqlServerUpdate(d *pluginsdk.ResourceData, meta interface{}) erro
 		},
 	}
 
-	if ok := d.HasChange("identity"); ok {
-		expandedIdentity, err := expandSqlServerIdentity(d.Get("identity").([]interface{}))
+	if v, ok := d.GetOk("identity"); ok {
```

Comment:
this should be using HasChange since this is a delta update method - the `else` statement below should be passing the existing identity instead

Reply:
After tested, I found `identity` has to be set when `primary_user_assigned_identity_id` is set. `primary_user_assigned_identity_id` is using `d.GetOk` so that it occurs the error user mentioned.